--- sqlite3-sql.lisp	2010-07-06 21:44:16.809767996 +0900
+++ sqlite3-sql.lisp.new	2010-07-06 21:43:39.509773797 +0900
@@ -320,6 +320,14 @@
     (or (string-equal ":memory:" name)
         (and (probe-file name) t))))
 
+(defmethod database-get-type-specifier ((type (eql 'integer))
+                                        args database
+                                        (db-type (eql :sqlite3)))
+  (declare (ignore database))
+  (if args
+      (format nil "INTEGER(~A)" (car args))
+      "INTEGER"))
+
 ;;; Database capabilities
 
 (defmethod db-type-has-boolean-where? ((db-type (eql :sqlite3)))
